src/Riok.Mapperly/Descriptors/MappingBuilders/ExplicitCastMappingBuilder.cs

```diff
@@ -16,6 +16,9 @@ public static class ExplicitCastMappingBuilder
         if (ctx.MapperConfiguration.UseDeepCloning && !ctx.Source.IsImmutable() && !ctx.Target.IsImmutable())
             return null;
 
+        if (ctx.Target.IsTupleType)
```

Comment:
Added comment:
ClassifyConversion does not check if tuple field member names are the same, if tuple check isn't done then `(A: int, B: int) -> (B: int, A: int)` would be mapped with:
`return ((int, int))source;` instead of `return (B: source.A, A: source.B);`